app/liquid_tags/codesandbox_tag.rb

```diff
@@ -24,39 +26,36 @@ def render(_context)
 
   private
 
-  def parse_id(input)
+  def parse_id_or_url_and_options(input)
+    match = pattern_match_for(input, [REGISTRY_REGEXP])
+    return [match[:video_id], parse_options(match[:options].split("&"))] if match
+
     id = input.split.first
     raise StandardError, "CodeSandbox Error: Invalid ID" unless valid_id?(id)
 
-    id
+    [id, parse_options(extract_options(input))]
   end
 
   def valid_id?(id)
-    id =~ /\A[a-zA-Z0-9\-]{0,60}\Z/
+    id =~ /\A[\w-]{,60}\Z/
   end
 
-  def parse_options(input)
+  def extract_options(input)
     _, *options = input.split
+    options
+  end
 
-    options.map { |option| valid_option(option) }.reject(&:nil?)
-
-    query = options.join("&")
+  def parse_options(options)
```

Comment:
@jeremyf @djuber Hey fellas, would you give my changes a quick once-over?

I removed the exception raised if invalid options are present.

I also added specs to properly cover this behavior.